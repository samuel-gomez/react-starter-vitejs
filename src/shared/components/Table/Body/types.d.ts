import { type ReactNode } from 'react';
import { type THeader } from '../Headers/Header';

export type Tcol = Omit<THeader, 'key'> & {
  hover?: string;
  children?: ReactNode;
  classModifier?: string;
};
