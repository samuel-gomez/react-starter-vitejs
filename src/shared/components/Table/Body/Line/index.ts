export { default } from './Line.container';
