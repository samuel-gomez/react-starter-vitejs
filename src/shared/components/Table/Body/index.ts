export { default } from './Body';
export type { TBody } from './Body';
