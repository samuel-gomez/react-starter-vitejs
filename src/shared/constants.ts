import packageJson from '../../package.json';

/** ***********************************************************************
 * FORM CONSTANTS
 ************************************************************************ */
export const MSG_REQUIRED = 'Le champ est obligatoire';

export const MSG_LENGTH_3 = 'Le champ doit faire au moins 3 caractères';
export const MSG_FORMAT = 'Format invalide';
export const MSG_ERROR_FORMAT = 'Format de fichier incorrect';

export const DEFAULT_OPTION_LABEL = '- Sélectionner -';
export const DEFAULT_OPTION = {
  value: '',
  label: DEFAULT_OPTION_LABEL,
};

// patterns
export const patternEmail = /^[\w-+]+@([\w-+]+\.)+[\w-]{2,}$/;
export const patternPhoneNumber = /^[+0][\d ]*$/;
export const patternZipCode = /^\d{5}$/;
export const patternSiret = /^\d{14}$/;
export const patternSiren = /^\d{9}$/;
export const patternSiretSiren = /^\d{9,14}$/;

export const phoneNumberRule = {
  required: true,
  minLength: 4,
  maxLength: 20,
  pattern: patternPhoneNumber,
};

export const emailRule = {
  required: true,
  minLength: 6,
  maxLength: 60,
  pattern: patternEmail,
};

/** ***********************************************************************
 * API STATUS CONSTANTS
 ************************************************************************ */
export const STATUS_HTTP = {
  SUCCESS: 200,
  CREATED: 201,
  BAD_REQUEST: 400,
  UNAUTHORIZE: 401,
  FORBIDDEN: 403,
  NOTFOUND: 404,
  SERVER_ERROR: 500,
};

export const STATUS_API = {
  SUCCESS: STATUS_HTTP.SUCCESS,
  CREATED: STATUS_HTTP.CREATED,
  WARNING: 40,
  ERROR: 50,
};

/** ***********************************************************************
 * API URL
 ************************************************************************ */
export const API_URL = {
  BASE: 'base',
  GITHUB: 'github',
};

export const STATUS_HTTP_MESSAGES = {
  [STATUS_HTTP.SUCCESS]: 'Succès: La requête a réussi',
  [STATUS_HTTP.CREATED]: 'Succès: Création réussie',
  [STATUS_HTTP.BAD_REQUEST]: 'Erreur: La syntaxe de la requête est erronée',
  [STATUS_HTTP.UNAUTHORIZE]: 'Erreur: Une authentification est nécessaire pour accéder à la ressource',
  [STATUS_HTTP.FORBIDDEN]: 'Erreur: Accès non autorisé',
  [STATUS_HTTP.NOTFOUND]: 'Erreur: Elément non trouvé',
  [STATUS_HTTP.SERVER_ERROR]: 'Erreur: Problème technique ! Contacter votre support',
};

/** ***********************************************************************
 * USER CONSTANTS
 ************************************************************************ */

export const ADMIN = 'Admin';
export const READER = 'Reader';
export const WRITER = 'Writer';
export const USER = 'User';
export const ALL = '';

export const PROFILS = [ADMIN, READER, WRITER, USER, ALL];

/** ***********************************************************************
 * COMMONS CONSTANTS
 ************************************************************************ */

export const MODIFIER_CLASS = {
  success: 'success',
  disabled: 'disabled',
};

/** ***********************************************************************
 * VERSION CONSTANT
 ************************************************************************ */

const VERSION_TOOLKIT = 'master';

/** ***********************************************************************
 * DEMO CONSTANTS
 ************************************************************************ */

export const STORYBOOK = 'https://axafrance.github.io/react-toolkit/latest/storybook/?path=/story/';
export const DESIGN_SYSTEM = 'https://axafrance.github.io/design-system/';
export const GITHUB = `https://github.com/AxaGuilDEv/react-toolkit/tree/v${packageJson.dependencies['@axa-fr/react-toolkit-all']}/packages/`;
export const GITHUB_API = `AxaGuilDEv/react-toolkit/${VERSION_TOOLKIT}/packages/`;
