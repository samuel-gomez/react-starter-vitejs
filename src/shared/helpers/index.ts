export { formatDate, isValidDate, setDate } from './formatDate';
export { default as isEmptyOrNull, isEmptyArray, isEmptyObject } from './isEmptyOrNull';
export { default as mergeObj } from './mergeObj';
export { default as setAnomalyEmptyItems } from './setAnomalyEmptyItems';
export { default as setConfirmClassModifier } from './setConfirmClassModifier';
export { default as emptyFunction } from './emptyFunction';
export { default as setRequiredFieldRule, setRequiredMessage } from './setRequiredFieldRule';
