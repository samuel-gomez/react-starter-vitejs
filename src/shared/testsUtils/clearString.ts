const cleanString = (str?: string) => str?.replace(/\s/g, '').replace(/\n/g, '').replace(/\t/g, '').trim();

export default cleanString;
