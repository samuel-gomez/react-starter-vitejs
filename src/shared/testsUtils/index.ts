export * from './customRender';
export { default as emptyFunction } from '../helpers/emptyFunction';
export { default as MOCK_API_URL } from './constants';
export { default as renderWithContainer } from './renderWithContainer';
