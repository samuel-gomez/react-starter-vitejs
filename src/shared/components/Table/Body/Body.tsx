import { ReactNode } from 'react';
import Line from './Line';

export type Tcol = {
  label?: string;
  hover?: string;
  children?: ReactNode;
  classModifier?: string;
};
export type TCols = {
  [k: string]: Tcol;
};

type TItems = {
  key: string;
  modifier?: string;
  cols: TCols;
};

export type TBody = {
  ariaLabel?: string;
  items?: TItems[];
  children?: ReactNode;
};

const Body = ({ items = [], children, ariaLabel = 'table-body' }: TBody) => (
  <tbody className="af-table__body" aria-label={ariaLabel}>
    {items.map(({ key, modifier, cols }) => (
      <Line key={key} modifier={modifier} cols={Object.entries({ ...cols })} />
    ))}
    {children}
  </tbody>
);

export default Body;
