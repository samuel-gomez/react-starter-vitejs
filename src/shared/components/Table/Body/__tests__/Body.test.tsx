import { describe, it, expect } from 'vitest';
import { renderWithContainer } from 'shared/testsUtils';
import Body from '../Body';

const container = document.createElement('table');

describe('Body', () => {
  it('Render <Body/> with empty items', () => {
    const { baseElement } = renderWithContainer(<Body />, container);
    expect(baseElement).toMatchSnapshot();
  });

  const items = [
    {
      key: '5f52a9d04ddf8b6c00052a0a',
      cols: {
        firstname: { label: 'Antonius' },
        lastname: { label: 'Panketh' },
        birthdate: { label: '20/12/1968' },
        sexe: { label: 'F' },
      },
    },
  ];

  it('Render <Body/> with 1 item', () => {
    const { baseElement } = renderWithContainer(<Body items={items} />, container);
    expect(baseElement).toMatchSnapshot();
  });
});
