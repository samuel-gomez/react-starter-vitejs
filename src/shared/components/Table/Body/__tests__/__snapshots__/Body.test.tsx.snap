// Vitest Snapshot v1

exports[`Body > Render <Body/> with 1 item 1`] = `
<table>
  <tbody
    aria-label="table-body"
    class="af-table__body"
  >
    <tr
      aria-label="table-body-line"
      class="af-table__tr"
    >
      <td
        class="af-table__cell"
      >
        Antonius
      </td>
      <td
        class="af-table__cell"
      >
        Panketh
      </td>
      <td
        class="af-table__cell"
      >
        20/12/1968
      </td>
      <td
        class="af-table__cell"
      >
        F
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Body > Render <Body/> with empty items 1`] = `
<table>
  <tbody
    aria-label="table-body"
    class="af-table__body"
  />
</table>
`;
